src/logic/BrazeMessages.tsx

```diff
@@ -3,15 +3,51 @@
 import type appboy from '@braze/web-sdk-core';
 import { MessageCache } from './LocalMessageCache';
 import type { SlotName } from './types';
-
-type Extras = Record<string, string>;
+import {
+    COMPONENT_NAME as APP_BANNER_NAME,
+    canRender as appBannerCanRender,
+} from '../AppBanner/canRender';
+import {
+    COMPONENT_NAME as G_IN_2020_BANNER_NAME,
+    canRender as gIn2020BannerCanRender,
+} from '../TheGuardianIn2020Banner/canRender';
+import {
+    COMPONENT_NAME as DIGITAL_SUBSCRIBER_APP_BANNER,
+    canRender as digitialSubCanRender,
+} from '../DigitalSubscriberAppBanner/canRender';
+import {
+    COMPONENT_NAME as SPECIAL_EDITION_BANNER,
+    canRender as specialEdCanRender,
+} from '../SpecialEditionBanner/canRender';
+
+export type Extras = Record<string, string>;
 export type ErrorHandler = (error: Error, identifier: string) => void;
 
 interface BrazeMessagesInterface {
     getMessageForBanner: () => Promise<BrazeMessage>;
     getMessageForEndOfArticle: () => Promise<BrazeMessage>;
 }
 
+const COMPONENT_CAN_RENDER_MAPPINGS: Record<
+    string,
+    (brazeMessageProps: Record<string, unknown>) => boolean
+> = {
+    [APP_BANNER_NAME]: appBannerCanRender,
+    [DIGITAL_SUBSCRIBER_APP_BANNER]: digitialSubCanRender,
+    [SPECIAL_EDITION_BANNER]: specialEdCanRender,
+    [G_IN_2020_BANNER_NAME]: gIn2020BannerCanRender,
+};
+
+export const canRenderBrazeMsg = (msgExtras: Extras | undefined): boolean => {
```

Comment:
I wonder if this function and the mappings etc should sit with `BrazeMessageComponent`? It feels somewhat analogous to the `canRender` function which sits with each component, but higher level. Maybe in `src/BrazeMessageComponent/canRender`?